src/client/Microsoft.Identity.Client/Platforms/iOS/BrokerKeyHelper.cs

```diff
@@ -44,7 +44,18 @@ internal static byte[] GetRawBrokerKey(ICoreLogger logger)
                     ValueData = byteData
                 };
 
-                var result = SecKeyChain.Add(record);
+                var result = SecKeyChain.Add(secRecord);
+
+                if (result == SecStatusCode.DuplicateItem)
+                {
+                    logger.Info(iOSBrokerConstants.FailedToSaveBrokerKey + result);
+                    result = SecKeyChain.Remove(record);
```

Comment:
>result = SecKeyChain.Remove(record); [](start = 20, length = 36)

we are only here if they key is null (after querying the data store). Will this really do anything for this case? Should this have been remove secRecord instead?

Reply:
For some reason, the customer is getting a dubplicate keychain error here, so even though the key is null, there is still something in the keychain. i wasn't able to repro this naturally, so am waiting for customer response.

---
In reply to: [378472489](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1627#discussion_r378472489) [](ancestors = 378472489)